pkg/runtime/cache/account.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package cache
+
+import (
+	"sync"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	informersv1 "k8s.io/client-go/informers/core/v1"
+	kubernetes "k8s.io/client-go/kubernetes"
+	k8scache "k8s.io/client-go/tools/cache"
+)
+
+const (
+	// ACKRoleAccountMap is the name of the configmap map object storing
+	// all the AWS Account IDs associated with their AWS Role ARNs.
+	ACKRoleAccountMap = "ack-role-account-map"
+)
+
+// AccountCache is responsible for caching the CARM configmap
+// data. It is listening to all the events related to the CARM map and
+// make the changes accordingly.
+type AccountCache struct {
+	sync.RWMutex
+
+	log logr.Logger
+
+	// ConfigMap informer
+	informer k8scache.SharedInformer
+	roleARNs map[string]string
+}
+
+// NewAccountCache makes a new AccountCache from a client.Interface
+// and a logr.Logger
+func NewAccountCache(clientset kubernetes.Interface, log logr.Logger) *AccountCache {
+	sharedInformer := informersv1.NewConfigMapInformer(
+		clientset,
+		currentNamespace,
+		informerResyncPeriod,
+		k8scache.Indexers{},
+	)
+	return &AccountCache{
+		informer: sharedInformer,
+		log:      log.WithName("AccountCache"),
+		roleARNs: make(map[string]string),
+	}
+}
+
+// resourceMatchACKRoleAccountConfigMap verifies if a resource is
+// the CARM configmap. It verifies the name, namespace and object type.
+func resourceMatchACKRoleAccountsConfigMap(raw interface{}) bool {
+	object, ok := raw.(*corev1.ConfigMap)
+	return ok && object.ObjectMeta.Name == ACKRoleAccountMap
+}
+
+// Run adds the default event handler functions to the SharedInformer and
+// runs the informer to begin processing items.
+func (c *AccountCache) Run(stopCh <-chan struct{}) {
+	c.informer.AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if resourceMatchACKRoleAccountsConfigMap(obj) {
+				object := obj.(*corev1.ConfigMap).DeepCopy()
+				c.log.V(1).Info("ack-role-account-map has been created")
+				c.updateAccountRoleData(object.Data)
+				c.log.V(1).Info("cached ack-role-account-map data")
+			}
+		},
+		UpdateFunc: func(old, new interface{}) {
```

Comment:
teeny nit: `new` is a keyword in Go and naming variables the same as keywords (something called "variable shadowing" ) is generally discouraged. Might be better to name these `orig` and `desired`?

Reply:
Ahh, nice catch! I will change that

Reply:
Done.